modcc/printer/cexpr_emit.cpp

```diff
@@ -21,8 +21,14 @@ std::ostream& operator<<(std::ostream& out, as_c_double wrap) {
     case FP_ZERO:
         return out << (neg? "-0.": "0.");
     default:
-        return out <<
-            (std::stringstream{} << io::classic << std::setprecision(17) << wrap.value).rdbuf();
+        double val;
+        std::stringstream s;
+        if (std::modf(wrap.value, &val) == 0) {
+            s << io::classic << std::fixed << std::setprecision(1) << wrap.value;
```

Comment:
Why `setprecision(1)`? A comment that explains what is going on would be helpful.

Reply:
`std::max` and `std::min` were complaining about comparing `value_type` and `int` because the generated expression was 
`max( 1, min(g[i_]+weight,  10));`.
We have the `as_c_double` struct which should print all numbers as doubles AFAIK, but it doesn't actually do that for `int`. `std::setprecision(1)` will print `1.0` and `10.0` instead of `1` and `10` which fixes the issue.

I'll add a comment explaining this.

Reply:
Thanks.